src/librbd/exclusive_lock/ReleaseRequest.cc

```diff
@@ -130,22 +131,52 @@ Context *ReleaseRequest<I>::handle_flush_notifies(int *ret_val) {
   ldout(cct, 10) << __func__ << dendl;
 
   assert(*ret_val == 0);
-  send_close_journal();
+
+  send_append_journal_event();
   return nullptr;
 }
 
 template <typename I>
-void ReleaseRequest<I>::send_close_journal() {
-  {
-    RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
-    std::swap(m_journal, m_image_ctx.journal);
-  }
+void ReleaseRequest<I>::send_append_journal_event(){
+    CephContext *cct = m_image_ctx.cct;
+    ldout(cct, 10) << __func__ << dendl;
+    {
+      RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
+      std::swap(m_journal, m_image_ctx.journal);
+    }
 
-  if (m_journal == nullptr) {
-    send_close_object_map();
-    return;
-  }
+    if (m_journal == nullptr) {
+       send_close_object_map();
+       return;
+    }
+
+    using klass = ReleaseRequest<I>;
+    Context *ctx = create_context_callback<
+      klass, &klass::handle_append_journal_event>(this);
+    //append journal event
+    uint64_t journal_op_tid = m_journal->allocate_op_tid();
+    {
+      RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
+      m_journal->append_op_event(journal_op_tid,journal::EventEntry(journal::ExclusiveLockReleaseEvent()),ctx);
```

Comment:
Add a new ```librbd::Journal::append_lock_release(Context *ctx)``` method and invoke it here.  That method would look similar to the following:

```
template <typename I>
void Journal<I>::append_lock_release_event(Context *ctx) {
  CephContext *cct = m_image_ctx.cct;
  ldout(cct, 10) << this << " " << __func__ << dendl;

  journal::EventEntry event_entry((journal:: ExclusiveLockReleaseEvent()));

  bufferlist bl;
  ::encode(event_entry, bl);

  Future future;
  {
    Mutex::Locker locker(m_lock);
    assert(m_state == STATE_READY);
    future = m_journaler->append(m_tag_tid, bl);
  }
  future.flush(ctx);
}
```

In test/librbd/mock/MockJournal.h, you'll need to add a new mock to match the new method signature:

```
MOCK_METHOD1(append_lock_release_event, void(Context *));
```

In test/librbd/exclusive_lock/test_mock_ExclusiveLock.cc, you'll need to set up the expectation that the ```append_lock_release_event``` method will be called:

```
  void expect_append_lock_release_event(MockImageCtx &mock_image_ctx,
                                        MockJournal &mock_journal, int r) {
    EXPECT_CALL(mock_journal, append_lock_release_event(_))
                  .WillOnce(CompleteContext(r, mock_image_ctx.image_ctx->op_work_queue));
  }
```

In each test, invoke the new ```expect_append_lock_release_event``` method after ```expect_flush_notifies``` (since that is the order in which the methods are invoked).